conftest.py

```diff
@@ -137,22 +148,71 @@ def pytest_runtest_makereport(item, call):
 
 # fixtures
 
-def setup_host(hostname_or_ip):
-    pool = Pool(hostname_or_ip)
-    h = pool.master
-    return h
-
 @pytest.fixture(scope='session')
 def hosts(pytestconfig):
+    nested_list = []
+
+    def setup_host(hostname_or_ip, *, config=None):
+        host_vm = None
+        if hostname_or_ip.startswith("cache://"):
+            nest_hostname = config.getoption("nest")
```

Comment:
There is no check to ensure that `config` is not `None`.
In fact `setup_host` is defined and used only inside `hosts()` and could as well use `pytestconfig` from `hosts()` instead of passing the config as parameter.

Reply:
Explicit check added